src/main/scala/firrtl/ir/IR.scala

```diff
@@ -328,7 +333,7 @@ class IntWidth(val width: BigInt) extends Width with Product {
     case w: IntWidth => width == w.width
     case _ => false
   }
-  override def hashCode = width.toInt
+  override lazy val hashCode = width.toInt
```

Comment:
I originally wanted to complain that this will lead to very clustered hashCodes (since width tends to be clustered) and a bad distribution. However, it seems scala map implementations (like HashMap) will 'improve' the hashCode anyway:
e.g. https://github.com/scala/scala/blob/v2.12.2/src/library/scala/collection/immutable/HashMap.scala#L86

so this might not be an issue. Although, it would be good to get a collections expert to affirm this.